portal-bridge/src/types/state/spec_id.rs

```diff
@@ -1,61 +1,106 @@
-pub const CANCUN_BLOCK_NUMBER: u64 = 19426587;
+pub const SPEC_FORK_BLOCK_NUMBER: [(SpecId, u64); 18] = [
+    (SpecId::FRONTIER, 0),
+    (SpecId::FRONTIER_THAWING, 200000),
+    (SpecId::HOMESTEAD, 1150000),
+    (SpecId::DAO_FORK, 1920000),
+    (SpecId::TANGERINE, 2463000),
+    (SpecId::SPURIOUS_DRAGON, 2675000),
+    (SpecId::BYZANTIUM, 4370000),
+    (SpecId::CONSTANTINOPLE, 7280000),
+    (SpecId::PETERSBURG, 7280000), // Petersburg is a network upgrade that replaced Constantinople
+    (SpecId::ISTANBUL, 9069000),
+    (SpecId::MUIR_GLACIER, 9200000),
+    (SpecId::BERLIN, 12244000),
+    (SpecId::LONDON, 12965000),
+    (SpecId::ARROW_GLACIER, 13773000),
+    (SpecId::GRAY_GLACIER, 15050000),
+    (SpecId::MERGE, 15537394),
+    (SpecId::SHANGHAI, 17034870),
+    (SpecId::CANCUN, 19426587),
+];
 
 pub fn get_spec_id(block_number: u64) -> SpecId {
-    if block_number >= CANCUN_BLOCK_NUMBER {
-        SpecId::CANCUN
-    } else if block_number >= SHANGHAI_BLOCK_NUMBER {
-        SpecId::SHANGHAI
-    } else if block_number >= MERGE_BLOCK_NUMBER {
-        SpecId::MERGE
-    } else if block_number >= GRAY_GLACIER_BLOCK_NUMBER {
-        SpecId::GRAY_GLACIER
-    } else if block_number >= ARROW_GLACIER_BLOCK_NUMBER {
-        SpecId::ARROW_GLACIER
-    } else if block_number >= LONDON_BLOCK_NUMBER {
-        SpecId::LONDON
-    } else if block_number >= BERLIN_BLOCK_NUMBER {
-        SpecId::BERLIN
-    } else if block_number >= MUIR_GLACIER_BLOCK_NUMBER {
-        SpecId::MUIR_GLACIER
-    } else if block_number >= ISTANBUL_BLOCK_NUMBER {
-        SpecId::ISTANBUL
-    } else if block_number >= PETERSBURG_BLOCK_NUMBER {
-        SpecId::PETERSBURG
-    } else if block_number >= CONSTANTINOPLE_BLOCK_NUMBER {
-        SpecId::CONSTANTINOPLE
-    } else if block_number >= BYZANTIUM_BLOCK_NUMBER {
-        SpecId::BYZANTIUM
-    } else if block_number >= SPURIOUS_DRAGON_BLOCK_NUMBER {
-        SpecId::SPURIOUS_DRAGON
-    } else if block_number >= TANGERINE_BLOCK_NUMBER {
-        SpecId::TANGERINE
-    } else if block_number >= DAO_FORK_BLOCK_NUMBER {
-        SpecId::DAO_FORK
-    } else if block_number >= HOMESTEAD_BLOCK_NUMBER {
-        SpecId::HOMESTEAD
-    } else if block_number >= FRONTIER_THAWING_BLOCK_NUMBER {
-        SpecId::FRONTIER_THAWING
-    } else {
-        SpecId::FRONTIER
+    for (spec_id, fork_block_number) in SPEC_FORK_BLOCK_NUMBER.iter().rev() {
+        if block_number >= *fork_block_number {
+            return *spec_id;
+        }
+    }
+    SpecId::LATEST
+}
+
+pub fn get_spec_block_number(spec_id: SpecId) -> u64 {
+    match SPEC_FORK_BLOCK_NUMBER
+        .iter()
+        .find(|(id, _)| *id == spec_id)
+        .map(|(_, block_number)| *block_number)
+    {
+        Some(block_number) => block_number,
+        None => unimplemented!("SpecId {:?} is not supported", spec_id),
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_get_spec_id() {
```

Comment:
nit: I would add these three tests for `get_spec_id`:
- block number is 1
- block number is after CANCUN and it returns CANCUN
- block number is between two forks, and it returns correct one (you don't need to check every pair of forks, but you can with a loop)